static/app/views/settings/project/dynamicSampling/modals/specificConditionsModal/index.tsx

```diff
@@ -367,10 +337,16 @@ export function SpecificConditionsModal({
               }}
               placeholder={'\u0025'}
               step="10"
-              value={samplePercentage}
+              value={samplePercentage ?? null}
+              flexibleControlStateSize
               inline={false}
-              hideControlState={!errors.sampleRate}
-              error={errors.sampleRate}
+              error={
+                sampleRateEdited && !validSampleRate
```

Comment:
Do we need the `sampleRateEdited` check?
I'd like us to show this validation error as soon as you open this modal with invalid rule.

Reply:
the user will experience this behaviour with the code from the [next PR](https://github.com/getsentry/sentry/pull/39983)